mleap-xgboost-runtime/src/main/scala/ml/combust/mleap/xgboost/runtime/XGBoostPredictorClassification.scala

```diff
@@ -0,0 +1,31 @@
+package ml.combust.mleap.xgboost.runtime
+
+import biz.k11i.xgboost.util.FVec
+import ml.combust.mleap.core.types.NodeShape
+import ml.combust.mleap.runtime.function.UserDefinedFunction
+import ml.combust.mleap.tensor.Tensor
+import ml.combust.mleap.runtime.frame.{MultiTransformer, Row, Transformer}
+import ml.combust.mleap.core.util.VectorConverters._
+import XgbConverters._
+
+
+case class XGBoostPredictorClassification(
+                                  override val uid: String = Transformer.uniqueName("xgboost.classification"),
+                                 override val shape: NodeShape,
+                                 override val model: XGBoostPredictorClassificationModel) extends MultiTransformer {
+  override val exec: UserDefinedFunction = {
+
+    // Since the Predictor is our performant implementation, we only compute probability for performance reasons.
+    val probability = shape.getOutput("probability").map {
+      _ => (data: FVec) => Some(model.predictProbabilities(data): Tensor[Double])
+    }.getOrElse((_: FVec) => None)
+
+    val f = (features: Tensor[Double]) => {
+      val data: FVec = features.asXGBPredictor
+      val rowData = Seq(probability(data).get)
```

Comment:
What if `probability(data)` returns a `None`? Calling `get` on a `None` object will throw exception. Or are you intentionally letting it throw the error rather than checking it beforehand?

Reply:
This file is copied from https://github.com/combust/mleap/blob/master/mleap-xgboost-runtime/src/main/scala/ml/combust/mleap/xgboost/runtime/XGBoostClassificationModel.scala and https://github.com/combust/mleap/blob/master/mleap-xgboost-runtime/src/main/scala/ml/combust/mleap/xgboost/runtime/XGBoostRegressionModel.scala so I'm using the same proven code that it's already been used for the xgboost4j regressor and classifier.

It's good for exceptions to _throw_ when something unexpected happens.

Reply:
Do you mean `XGBoostClassification` here? https://github.com/combust/mleap/blob/master/mleap-xgboost-runtime/src/main/scala/ml/combust/mleap/xgboost/runtime/XGBoostClassification.scala#L36

It is filtering by `_.isDefined` first and then calling `_.get`, which will not cause any exceptions. However, directly calling `get` from an `Option` object is not ideal. The error log will just say something like `java.util.NoSuchElementException: None.get`, which is not informant for the users at all I think?

Reply:
Oh yes, I removed the _isDefined_ filter because I don't think we should ever expect non-defined here.  "None" is really unexpected because XGBoost Predictor should always have "probability" defined, shouldn't it?  So it's good to fail fast without masking the exception to the user (I'd be happier to wrap any expected user-facing exceptions).

Even in the original code, if .isDefined was false, that value would be _.filtered_ out and the _Row_ would end up with the values all shifted left by the number of missing fields, which would break the output schema. I might be missing something here but I am not sure if isDefined makes sense. Unless it's just the norm to put it by default when accessing a scala Option object?